src/animated.tpp

```diff
@@ -16,244 +16,224 @@
+		frames_.push_back(frame{duration, value, last_frame.start_time_ + last_frame.duration_});
 	}
 }
 
+
+// ============================================================================
+// Playback Control
+// ============================================================================
+
 template<typename T>
-inline void animated<T>::start_animation(const std::chrono::milliseconds& start_time, bool cycles)
+inline void animated<T>::start_animation(const std::chrono::milliseconds& start_time_offset, bool cycles)
 {
 	started_ = true;
-	last_update_tick_ = get_current_animation_tick();
-	acceleration_ = 1.0; // assume acceleration is 1, this will be fixed at first update_last_draw_time
-	start_tick_ = last_update_tick_ + (starting_frame_time_ - start_time);
+	paused_ = false;
 	cycles_ = cycles;
-	current_frame_key_ = 0;
-	force_next_update_ = !frames_.empty();
-}
-
-template<typename T>
-inline void animated<T>::update_last_draw_time(double acceleration)
-{
-	if(acceleration > 0 && acceleration_ != acceleration) {
-		auto tmp = tick_to_time(last_update_tick_);
-		acceleration_ = acceleration;
-		start_tick_ = last_update_tick_ + std::chrono::duration_cast<std::chrono::milliseconds>((starting_frame_time_ - tmp) / acceleration_);
-	}
-
-	if(!started_ && start_tick_ != std::chrono::steady_clock::time_point{}) {
-		// animation is paused
-		start_tick_ += get_current_animation_tick() - last_update_tick_;
-	}
-
-	last_update_tick_ = get_current_animation_tick();
-	if(force_next_update_) {
-		force_next_update_ = false;
-		return;
-	}
-
-	if(does_not_change_) {
-		return;
-	}
-
-	// Always update last_update_tick_, for the animation_time functions to work.
-	if(!started_) {
-		return;
+	current_frame_index_ = 0;
+	const auto current_tick = get_current_animation_tick(uses_acceleration_);
+	animation_start_tick_ = current_tick - start_time_offset;
+	if (!frames_.empty()) {
+		next_frame_tick_ = animation_start_tick_ + frames_[0].duration_;
 	}
-
-	if(frames_.empty()) {
-		does_not_change_ = true;
-		return;
-	}
-
-	// Check if it's time to move on to the next frame. Might skip multiple frames
-	// if the animation was paused off screen and has now scrolled back on screen.
-	auto animation_time = get_animation_time();
-	const auto end_time = get_end_time();
-	if(cycles_ && animation_time > end_time) {
-		const auto time_in_current_cycle = (animation_time - end_time) % get_animation_duration();
-		const auto time_to_skip = animation_time - time_in_current_cycle;
-
-		// Reset start tick to the beginning of the current cycle.
-		start_tick_ += std::chrono::floor<std::chrono::milliseconds>(time_to_skip / acceleration_);
-
-		// We could be anywhere in the cycle. Assume first frame and adjust as needed below.
-		current_frame_key_ = 0;
-		animation_time = time_in_current_cycle;
-	}
-
-	// Update the index key to the appropriate frame for the current animation time.
-	auto iter = frames_.begin() + current_frame_key_;
-	while(std::next(iter) != frames_.end()) {
```

Comment:
Where did this code go? We spent a ton of time fixing the perf issue related to it...